c/unixFileService.c

```diff
@@ -981,6 +981,43 @@ static int serveUnixFileChangeOwner (HttpService *service, HttpResponse *respons
   return 0;
 }
 
+static int serveUnixFileChangeTag (HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  char *routeFileFrag = stringListPrint(request->parsedFile, 2, 1000, "/", 0);
+
+  if (routeFileFrag == NULL) {
+   respondWithError(response, HTTP_STATUS_BAD_REQUEST, "File name not provided");
```

Comment:
Is this the right message? Shouldn't this be "Allocation failed" with `HTTP_STATUS_INTERNAL_SERVER_ERROR`? Same true for the next 2 calls.

Reply:
isn't it a bad request with the required params. I am not sure about this normally if a required param is not there it should be a bad request

Reply:
This sort of logic is used in a few places for checking if the url has the right shape 
https://github.com/zowe/zss/blob/master/c/rasService.c#L123
Seems like this can be the case when the stringlistprint returns with pos=0
https://github.com/zowe/zowe-common-c/blob/master/c/utils.c#L1681

also anyone know what this is https://github.com/zowe/zowe-common-c/blob/master/c/utils.c#L1658
seems like a weird thing to do.

Reply:
From what I can see in `stringListPrint`, it would only return NULL if `SLHAlloc` failed. However, even if it fails we will segfault because the `stringListPrint` code doesn't check the `SLHAlloc` return value. In case at some point in the future we add a NULL check inside `stringListPrint`, the callers of this function will also need to check the return value for NULL. So it'd be reasonable to start doing that now. If we want to fix all the places at ones in the future, that's ok, but the messages is still misleading.

Reply:
@ifakhrutdinov can we meege this PR

Reply:
@Akayeshmantha 
The NULL check at the top has the wrong message and passing an empty path isn't really handled.

![image](https://user-images.githubusercontent.com/34161978/89173768-0fc29b00-d585-11ea-9ac5-85f4061deed6.png)

Reply:
Yes @ifakhrutdinov 😃.It is not handled my bad 😞

Reply:
edge cases when absolute path not provided.
![Screenshot from 2020-08-03 20-10-52](https://user-images.githubusercontent.com/20354461/89221587-cfd6d480-d5d3-11ea-988b-9bb83cd2cc00.png)

Reply:
edge cases when absolute path is not provided.

![Screenshot from 2020-08-03 21-51-40](https://user-images.githubusercontent.com/20354461/89221648-e8df8580-d5d3-11ea-8708-8c622766c680.png)

Reply:
@ifakhrutdinov  please have a look

Reply:
The NULL check is not correct since the result can only be NULL when the SLH fails which has nothing to do with this particular service. Please either remove it or use the following approach:

```c
  char *routeFileFrag = stringListPrint(request->parsedFile, 2, 1000, "/", 0);
  if (routeFileFrag == NULL) {
   respondWithError(response, "Allocation error", HTTP_STATUS_INTERNAL_SERVER_ERROR);
   return 0;
  }

  if (strlen(routeFileFrag) == 0) {
   respondWithJsonError(response, "Required absolute path of the resource is not provided", 
                        HTTP_STATUS_BAD_REQUEST, "Bad Request");
   return 0;
  }

  char *encodedRouteFileName = stringConcatenate(response->slh, "/", routeFileFrag);
  if (encodedRouteFileName == NULL) {
   respondWithError(response, "Allocation error", HTTP_STATUS_INTERNAL_SERVER_ERROR);
   return 0;
  }
  char *routeFileName = cleanURLParamValue(response->slh, encodedRouteFileName);
  if (routeFileName == NULL) {
   respondWithError(response, "Allocation error", HTTP_STATUS_INTERNAL_SERVER_ERROR);
   return 0;
  }
```

Reply:
sure @ifakhrutdinov . Yeah there is no null case